# Feature Extractions

##### Import Library

we need to import some library like pandas for handle dataframe, Customtokenizer (that we have created) to make the tokens, and Vectorizer that we want to have TF-IDF matrix.

In [2]:
import pandas as pd
from CustomTokenizer import Tokenizer
from Vectorizer import Vectorizer

Load the (sentences)feature

In [3]:
dataset = pd.read_csv('../Dataset/datasets.csv')
sentences = dataset['sentence']

In [7]:
# Instantiate the tokenizer
tokenizer = Tokenizer()
# fit and transform the sentences to the tokenizer.
str_tokens = tokenizer.fit_transform(sentences)
# Now we have tokens.

In [8]:
str_tokens

['กล้า ป่าว ระบบ การศึกษา มันดี อ่ะ ถุ้ย เฟส ปลอม สัส',
 'เบื่อ เวลา โพ สต แบบนี้ ชอบ โปรไฟล์ โปรไฟล์ ตัวเอง เสือก ลึก แบบนี้ จุ',
 'พ่อ ติ่ง รัฐบาล หรอ สัส ตัง แพง แดก อะ ควย',
 'ลบ ไอ้ ออกจาก โพสต์ เกรียน กวน ส้นตีน แน่จริง กล้า เฟส โพสต์ ไหม',
 'เครียด อะ ตอนนี้',
 'กำลังใจ สู้',
 'ท้อ ไหว 😭😭',
 'เรียน เหมือน เครียด หนัก 😭😭😭 คน เรียน',
 'หดหู่ ทีม เรียน',
 'เศร้า สอบ มีโอกาส แก้ตัว รอบ ต้องโทษ ตัวเอง ทำ ออกมา ดี',
 'เสียกำลังใจ',
 'พี่ กำลังใจ สู้',
 'ดีใจ',
 'ชีวิต สำเร็จ ความพยายาม เข้มแข็ง เดินทาง ก้อ เกิดขึ้น สู้ นะคะ',
 'ขอบคุณ กำลังใจ นะคะ ทำ ดี',
 'สู้',
 'สู้ ยังมี หนทาง ผม กำลังใจ',
 'สู้ กะ อี อย่า ท้อ เดิน ข้างหน้า ต่อไป',
 'เรียน รัก ชอบ ทำ ได้ดี มีความสุข',
 'สู้ ชีวิต การต่อสู้ 😊✌',
 'เว็บ ล่ม ล่ม เหมือน อนาคต',
 'หัว ร้อน ตอนนี้',
 'เย สเเม่ ไอ้ เปรต ต่อย ที่มา หัว ดอ เเร้วเ เละ เห งัน',
 'ตรรกะ ป่วย ป่ะ',
 'ระบบ เน่า ทปอ สักที เห้อ',
 'เรื่อง ดีใจ สอบ ติด ค้นพบ ตัวเอง',
 'มาจาก นคร ไปหา เจ้ ยย ดีใจ เจอ นะค้า าา หนู สอบ ติด 💕💕 กอด',
 'ทำได้ เย้',
 'ติด เเล้ว',
 'ยินดี

TF-IDF matrix

In [13]:
# Instantiate (Tfidf)vectorizer object.
vectorizer = Vectorizer()
# fit the sentences to Create TF-IDF.
vectorizer.fit(str_tokens)

Vectorizer()

##### Explore the tokens, Vocabulary and The TF-IDF matrix

In [10]:
vectorizer.vocabulary_

{'กล้า': 25,
 'ป่าว': 403,
 'ระบบ': 524,
 'การศึกษา': 39,
 'มันดี': 467,
 'อ่ะ': 777,
 'ถุ้ย': 286,
 'เฟส': 877,
 'ปลอม': 395,
 'สัส': 657,
 'เบื่อ': 858,
 'เวลา': 909,
 'โพ': 1047,
 'สต': 621,
 'แบบนี้': 991,
 'ชอบ': 177,
 'โปรไฟล์': 1045,
 'ตัวเอง': 257,
 'เสือก': 924,
 'ลึก': 576,
 'จุ': 170,
 'พ่อ': 446,
 'ติ่ง': 270,
 'รัฐบาล': 532,
 'หรอ': 714,
 'ตัง': 250,
 'แพง': 996,
 'แดก': 973,
 'อะ': 760,
 'ควย': 98,
 'ลบ': 568,
 'ไอ้': 1106,
 'ออกจาก': 758,
 'โพสต์': 1049,
 'เกรียน': 791,
 'กวน': 26,
 'ส้นตีน': 696,
 'แน่จริง': 987,
 'ไหม': 1100,
 'เครียด': 806,
 'ตอนนี้': 240,
 'กำลังใจ': 43,
 'สู้': 690,
 'ท้อ': 321,
 'ไหว': 1103,
 '😭😭': 1121,
 'เรียน': 892,
 'เหมือน': 930,
 'หนัก': 699,
 '😭😭😭': 1122,
 'คน': 86,
 'หดหู่': 697,
 'ทีม': 307,
 'เศร้า': 913,
 'สอบ': 646,
 'มีโอกาส': 485,
 'แก้ตัว': 961,
 'รอบ': 520,
 'ต้องโทษ': 282,
 'ทำ': 299,
 'ออกมา': 759,
 'ดี': 217,
 'เสียกำลังใจ': 914,
 'พี่': 441,
 'ดีใจ': 222,
 'ชีวิต': 188,
 'สำเร็จ': 671,
 'ความพยายาม': 109,
 'เข้มแข็ง': 799,
 'เดิ

Let's see the matrix. In this case it hard to see all of the values

In [11]:
transform_data = vectorizer.transform(sentences)
pd.DataFrame(transform_data.toarray())

0     1     2     3     4     5     6     7     8     9     ...  1118  \
0     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
1     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
2     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
3     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
4     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
..    ...   ...   ...   ...   ...   ...   ...   ...   ...   ...  ...   ...   
337   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
338   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
339   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
340   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   
341   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   0.0   

     1119  1120  1121  1122  1123  1124  1125  1126  1127  
0     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
1     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
2     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
3     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
4     0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
..    ...   ...   ...   ...   ...   ...   ...   ...   ...  
337   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
338   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
339   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
340   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  
341   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  

[342 rows x 1128 columns]

From this results aboved we have:
* 342 raws that refers to sentences that just fitted in the tokenizer and vectorizer.
* 1128 columns this refers to words/tokens that we have (in the vocabulary/TF-IDF matrix).